File: lobby-server/src/main/java/iieiiergn/smpAuth/lobby/GuideCommand.java
package iieiiergn.smpAuth.lobby;

import net.kyori.adventure.inventory.Book;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.event.ClickEvent;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextDecoration;
import net.minestom.server.command.builder.Command;
import net.minestom.server.entity.Player;

/** {@code /guide} — opens a written book covering /login, /verify and oh-my-smp server rules. */
public final class GuideCommand extends Command {

    public GuideCommand(Book book) {
        super("guide", "rules");

        setDefaultExecutor((sender, context) -> {
            if (!(sender instanceof Player player)) {
                sender.sendMessage(Component.text("플레이어만 사용할 수 있습니다."));
                return;
            }
            show(player, book);
        });
    }

    /** Opens the guide book for {@code player}; also used to auto-show it on first spawn. */
    static void show(Player player, Book book) {
        player.openBook(book);
        // ClickEvent.suggestCommand has no effect inside written books (Mojang restriction),
        // so offer the working button via chat instead.
        player.sendMessage(Component.text("키를 입력하려면 클릭하세요: ", NamedTextColor.GRAY)
                .append(Component.text("[/verify 입력하기]", NamedTextColor.BLUE, TextDecoration.UNDERLINED)
                        .clickEvent(ClickEvent.suggestCommand("/verify "))));
    }
}

File: smp-server/src/main/kotlin/iieiiergn/ohMySmp/guide/GuideCommand.kt
package iieiiergn.ohMySmp.guide

import iieiiergn.ohMySmp.config.PluginConfig
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

/** `/guide` — config.yml에 반영된 현재 서버 규칙 안내서를 펼친다. */
class GuideCommand(config: PluginConfig, nametagActive: Boolean) : CommandExecutor {

    private val book: ItemStack = GuideBook.build(config, nametagActive)

    override fun onCommand(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<out String>,
    ): Boolean {
        if (sender !is Player) {
            sender.sendMessage(Component.text("플레이어만 사용할 수 있습니다.", NamedTextColor.RED))
            return true
        }
        sender.openBook(book)
        return true
    }
}
